resources/acsl/datastructures.md

```diff
@@ -40,32 +56,169 @@
+
+Note that if you were to switch out the positions of the letters in *AMERICAN* to become something like *MCEAANIR*, the tree would look very different.
+
 ### Deleting Nodes
 
+In the case that a node needs to be deleted, this generally involves some minor shifting. Less, equal, and greater relationships between nodes still have to be considered for proper placement.
+
+| Original <img width=300/> | Deletion <img width=300/> | Description <img width=150/> |
+| --- | --- | --- |
+| | <img src="http://www.categories.acsl.org/wiki/images/thumb/8/8a/Bst-american-del-i.svg/300px-Bst-american-del-i.svg.png" width="150" height="150" /> | *I* has been removed. Since it had *no* children, no shifts needed to be made.
+| <img src="http://www.categories.acsl.org/wiki/images/thumb/b/bf/Bst-american.svg/300px-Bst-american.svg.png" width="150" height="150" /> | <img src="http://www.categories.acsl.org/wiki/images/thumb/3/39/Bst-american-del-r.svg/300px-Bst-american-del-r.svg.png" width="150" height="150" /> | *R* has been removed. Since it had *1* child, that child was moved up to be on the right of *M*. Since *N* comes after *M*, this shift is valid. |
+| | <img src="http://www.categories.acsl.org/wiki/images/thumb/3/3c/Bst-american-del-m.svg/300px-Bst-american-del-m.svg.png" width="150" height="150" /> | *M* has been removed. So, the left branch of *M* is moved up to take *M*'s place. *M*'s right branch is placed to the right of that branch. |            
+
 ### Searching for Nodes
 
-## Other Types of Trees
+To search for a specific code, refer to the following pseudocode algorithm:
+
+```code
+p = root
+found = FALSE
+while (p ≠ NIL) and (not found)
+  if (x < p’s key)
+    p = p’s left child
+  else if (x > p’s key)
+    p = p’s right child
+  else
+    found = TRUE
+  end if
+end while
+```
+
+So, searching first starts at the root of the tree and works its way down, moving either left or right based on what character you are searching for and where it is positioned within the tree. As an example, I will once again use the *AMERICAN* tree and attempt to find *E*.
+
+<img src="http://www.categories.acsl.org/wiki/images/thumb/b/bf/Bst-american.svg/300px-Bst-american.svg.png" width="150" height="150" />
+
+First, *p* is set to the root, *A*. *x* holds the value we are trying to find, *E*. 
+
+Then, we move on to the while loop, whose conditions have both been met. In this loop, since *E* > *p*'s key, *p* is then set to equal *M*. The cycle then ends, and we start the loop anew. This time, *E* < *p*'s key, so *p* is now set to equal *E*. Note that the while loop does not stop here! Instead, it would traverse through the loop once again and then execute the "else" statement to make *found = TRUE*. 
+
+So, by following this pseudocode, we have now successfully found our desired node.
+
+## Classifying Trees Further
+
+We've thoroughly worked with the *AMERICAN* binary search tree long enough, and now let's analyze it a bit more just for good measure. 
```

Comment:
Repetitive lol

Reply:
Did you mean my word choice? It is a bit wonky, so I changed it to "We've worked with the *AMERICAN* binary search tree for quite a while, and now let's analyze it a bit more just for good measure" now.

Reply:
Did you mean my word choice? It is a bit wonky, so I changed it to "We've worked with the *AMERICAN* binary search tree for quite a while, and now let's analyze it a bit more just for good measure" now.

Reply:
Did you mean my word choice? It is a bit wonky, so I changed it to "We've worked with the *AMERICAN* binary search tree for quite a while, and now let's analyze it a bit more just for good measure" now.